platform-tests/src/test/java/org/junit/platform/suite/engine/SuiteEngineTests.java

```diff
@@ -467,6 +476,25 @@ void passesOutputDirectoryProviderToEnginesInSuite() {
 		assertThat(outputDir).isDirectoryRecursivelyContaining("glob:**/test.txt");
 	}
 
+	@Test
+	void suiteEnginePassesRequestLevelStoreToSuiteTestDescriptors() {
+		UniqueId engineId = UniqueId.forEngine(SuiteEngineDescriptor.ENGINE_ID);
+		SuiteEngineDescriptor engineDescriptor = new SuiteEngineDescriptor(engineId);
+
+		SuiteTestDescriptor mockDescriptor = mock(SuiteTestDescriptor.class);
+		engineDescriptor.addChild(mockDescriptor);
+
+		EngineExecutionListener listener = mock(EngineExecutionListener.class);
+		NamespacedHierarchicalStore<Namespace> requestLevelStore = NamespacedHierarchicalStoreProviders.dummyNamespacedHierarchicalStore();
+
+		ExecutionRequest request = ExecutionRequest.create(engineDescriptor, listener,
+			mock(ConfigurationParameters.class), mock(OutputDirectoryProvider.class), requestLevelStore);
+
+		new SuiteTestEngine().execute(request);
+
+		verify(mockDescriptor).execute(listener, requestLevelStore);
```

Comment:
I think we should make the assertion a bit stronger to avoid accidentally depending on any `equals()` implementation.

```suggestion
		verify(mockDescriptor).execute(same(listener), same(requestLevelStore));
```